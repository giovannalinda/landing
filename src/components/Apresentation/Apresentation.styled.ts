import styled from 'styled-components'
import { theme } from '~/styles'

export const Container = styled.section`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  max-width: ${theme.grid.lg};
  width: 100%;
  margin: 12rem auto 0;
  height: calc(100vh - 20rem);
  padding-bottom: ${theme.space[8]};
`

export const TextGroup = styled.div`
  h1 {
    font-size: 7rem;
    font-weight: ${theme.font.weigths.regular};
    max-width: 70.7rem;
    position: relative;

    &::before {
      content: '';
      position: absolute;
      width: calc(100% - 14rem);
      height: 3.6rem;
      background: ${theme.colors.secondaryPurple};
      opacity: 0.15;
      left: 0;
      bottom: 0;
      z-index: 0;
      margin-left: ${theme.space[4]};
    }
  }

  span {
    color: ${theme.colors.secondaryPurple};
  }

  p {
    font-size: ${theme.font.sizes.paragraph};
    max-width: 55rem;
    color: ${theme.colors.gray500};
    margin: ${theme.space[6]} 0;
    line-height: ${theme.font.lineHeights.paragraph};
  }
`

export const ButtonGroup = styled.div`
  display: flex;
  align-items: center;

  button + button {
    margin-left: ${theme.space[4]};
  }
`

export const SocialNetworks = styled.div`
  p {
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.paragraph};
    margin-bottom: ${theme.space[2]};
  }

  button + button {
    margin-left: ${theme.space[2]};
  }
`
