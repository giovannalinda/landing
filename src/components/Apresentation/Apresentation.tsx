import Link from 'next/link'
import { ImGithub, ImLinkedin2 } from 'react-icons/im'
import { BsMedium, BsInstagram } from 'react-icons/bs'

import { motion } from 'framer-motion'
import { useTranslation } from '~/lib/useTranslation'
import { Button } from '~/components'

import * as S from './Apresentation.styled'

export function Apresentation() {
  const { t } = useTranslation()

  return (
    <S.Container>
      <S.TextGroup>
        <h1>
          {t('Hello, Iâ€™m Gio,')}
          <br />
          <motion.div animate={{ scale: [1, 1.2, 1] }}>
            <span>Front End Developer.</span>
          </motion.div>
        </h1>
        <p>
          {t(
            "A Front End Developer with a sparkle in her eye, who cares about every detail when building interfaces valuing the experience so that it positively impacts people's lives.",
          )}
        </p>

        <S.ButtonGroup>
          <Link href='mailto:eugiovannasouza@gmail.com'>
            <Button>{t('Hire me')}</Button>
          </Link>

          <Link href='#works'>
            <Button variant='white'>{t('View work')}</Button>
          </Link>
        </S.ButtonGroup>
      </S.TextGroup>

      <S.SocialNetworks>
        <small>{t('Follow me on')}</small>
        <motion.div animate={{ scale: [1, 1.1, 1] }}>
          <S.SocialNetworksGroup>
            <Button
              as='a'
              size='sm'
              variant='white'
              target='_blank'
              rel='noreferrer'
              href='https://github.com/giovannalinda'
            >
              <ImGithub size={20} />
            </Button>
            <Button
              as='a'
              size='sm'
              target='_blank'
              rel='noreferrer'
              href='https://www.linkedin.com/in/giovannalinda'
            >
              <ImLinkedin2 size={20} />
            </Button>
            <Button
              as='a'
              size='sm'
              variant='white'
              target='_blank'
              rel='noreferrer'
              href='https://www.instagram.com/_gripada'
            >
              <BsInstagram size={20} />
            </Button>
            <Button
              as='a'
              size='sm'
              variant='white'
              target='_blank'
              rel='noreferrer'
              href='https://medium.com/@eugiovannasouza'
            >
              <BsMedium size={20} />
            </Button>
          </S.SocialNetworksGroup>
        </motion.div>
      </S.SocialNetworks>
    </S.Container>
  )
}
