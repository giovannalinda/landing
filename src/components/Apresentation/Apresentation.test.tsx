import { render } from '~/__helpers__/app-tests'
import { Apresentation } from './Apresentation'

describe('Apresentation', () => {
  it('should match snapshot', () => {
    const { container } = render(<Apresentation />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
