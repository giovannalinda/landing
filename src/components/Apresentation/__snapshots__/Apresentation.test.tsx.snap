// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apresentation should match snapshot 1`] = `
<section
  class="sc-dkzDqf bQTnlx"
>
  <div
    class="sc-hKMtZM khMYUV"
  >
    <h1>
      Hello, I’m John, a 
      <span>
        Software Engineer.
      </span>
    </h1>
    <p>
      Product Designer, UI/UX Designer, and developer based in Brazil. Over the past 17 years, as an art director and designer, I’ve worked with big companies and up-and-coming startups.
    </p>
    <div
      class="sc-eCYdqJ gZZGbX"
    >
      <button
        class="sc-bczRLJ bRsxPm"
      >
        HIRE ME
      </button>
      <button
        class="sc-bczRLJ jFxIhk"
      >
        VIEW WORK
      </button>
    </div>
  </div>
  <div
    class="sc-jSMfEi jSHbGK"
  >
    <p>
      Follow me on
    </p>
    <button
      class="sc-bczRLJ gjpLmn"
    >
      <svg
        fill="currentColor"
        height="20"
        stroke="currentColor"
        stroke-width="0"
        version="1.1"
        viewBox="0 0 16 16"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 16c-4.412 0-8-3.588-8-8s3.587-8 8-8c4.412 0 8 3.587 8 8s-3.588 8-8 8v0zM14.747 9.094c-0.234-0.075-2.116-0.634-4.256-0.291 0.894 2.456 1.256 4.456 1.328 4.872 1.531-1.037 2.625-2.678 2.928-4.581v0zM10.669 14.3c-0.103-0.6-0.497-2.688-1.456-5.181-0.016 0.006-0.031 0.009-0.044 0.016-3.856 1.344-5.241 4.016-5.362 4.266 1.159 0.903 2.616 1.444 4.194 1.444 0.947 0 1.85-0.194 2.669-0.544v0zM2.922 12.578c0.156-0.266 2.031-3.369 5.553-4.509 0.088-0.028 0.178-0.056 0.269-0.081-0.172-0.388-0.359-0.778-0.553-1.159-3.409 1.022-6.722 0.978-7.022 0.975-0.003 0.069-0.003 0.138-0.003 0.209 0 1.753 0.666 3.356 1.756 4.566v0zM1.313 6.609c0.306 0.003 3.122 0.016 6.319-0.831-1.131-2.013-2.353-3.706-2.534-3.953-1.913 0.903-3.344 2.666-3.784 4.784v0zM6.4 1.366c0.188 0.253 1.431 1.944 2.55 4 2.431-0.909 3.459-2.294 3.581-2.469-1.206-1.072-2.794-1.722-4.531-1.722-0.55 0.003-1.088 0.069-1.6 0.191v0zM13.291 3.691c-0.144 0.194-1.291 1.663-3.816 2.694 0.159 0.325 0.313 0.656 0.453 0.991 0.050 0.119 0.1 0.234 0.147 0.353 2.275-0.284 4.534 0.172 4.759 0.219-0.016-1.612-0.594-3.094-1.544-4.256v0z"
        />
      </svg>
    </button>
    <button
      class="sc-bczRLJ jmfgsd"
    >
      <svg
        fill="currentColor"
        height="20"
        stroke="currentColor"
        stroke-width="0"
        version="1.1"
        viewBox="0 0 16 16"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.641 3.206c0.472 0 0.897 0.041 1.284 0.125 0.388 0.081 0.716 0.219 0.994 0.406 0.275 0.188 0.487 0.438 0.644 0.75 0.15 0.309 0.225 0.697 0.225 1.156 0 0.497-0.112 0.909-0.338 1.241-0.228 0.331-0.559 0.6-1.003 0.813 0.606 0.175 1.053 0.481 1.353 0.916 0.3 0.438 0.444 0.963 0.444 1.581 0 0.5-0.097 0.928-0.287 1.291-0.194 0.366-0.456 0.662-0.778 0.891-0.325 0.231-0.7 0.4-1.119 0.509-0.416 0.109-0.844 0.166-1.287 0.166h-4.772v-9.844h4.641zM4.359 7.181c0.384 0 0.703-0.091 0.953-0.275 0.25-0.181 0.369-0.481 0.369-0.894 0-0.228-0.041-0.419-0.122-0.566-0.084-0.147-0.194-0.263-0.334-0.344-0.138-0.084-0.294-0.141-0.478-0.172-0.178-0.034-0.366-0.050-0.556-0.050h-2.025v2.3h2.194zM4.478 11.372c0.213 0 0.416-0.019 0.606-0.063 0.194-0.044 0.366-0.109 0.509-0.209 0.144-0.097 0.266-0.225 0.353-0.394 0.088-0.166 0.128-0.378 0.128-0.637 0-0.506-0.144-0.869-0.428-1.088-0.284-0.216-0.662-0.322-1.131-0.322h-2.35v2.709h2.313z"
        />
        <path
          d="M11.331 11.338c0.294 0.287 0.716 0.431 1.266 0.431 0.394 0 0.738-0.1 1.022-0.3s0.456-0.412 0.522-0.631h1.725c-0.278 0.859-0.697 1.469-1.272 1.838-0.566 0.369-1.259 0.556-2.063 0.556-0.563 0-1.066-0.091-1.519-0.269-0.453-0.181-0.831-0.434-1.15-0.766-0.309-0.331-0.553-0.725-0.725-1.188-0.169-0.459-0.256-0.969-0.256-1.519 0-0.534 0.088-1.031 0.262-1.491 0.178-0.463 0.422-0.859 0.747-1.194s0.706-0.6 1.156-0.794c0.447-0.194 0.941-0.291 1.488-0.291 0.603 0 1.131 0.116 1.584 0.353 0.45 0.234 0.822 0.55 1.113 0.944s0.497 0.847 0.625 1.353c0.128 0.506 0.172 1.034 0.137 1.588h-5.147c0 0.559 0.188 1.094 0.484 1.378zM13.578 7.594c-0.231-0.256-0.628-0.397-1.106-0.397-0.313 0-0.572 0.053-0.778 0.159-0.203 0.106-0.369 0.237-0.497 0.394-0.125 0.156-0.213 0.325-0.262 0.503-0.050 0.172-0.081 0.331-0.091 0.469h3.188c-0.047-0.5-0.219-0.869-0.453-1.128z"
        />
        <path
          d="M10.444 4h3.991v0.972h-3.991v-0.972z"
        />
      </svg>
    </button>
    <button
      class="sc-bczRLJ gjpLmn"
    >
      <svg
        fill="currentColor"
        height="20"
        stroke="currentColor"
        stroke-width="0"
        version="1.1"
        viewBox="0 0 16 16"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.5 3h2.5v-3h-2.5c-1.93 0-3.5 1.57-3.5 3.5v1.5h-2v3h2v8h3v-8h2.5l0.5-3h-3v-1.5c0-0.271 0.229-0.5 0.5-0.5z"
        />
      </svg>
    </button>
    <button
      class="sc-bczRLJ gjpLmn"
    >
      <svg
        fill="currentColor"
        height="20"
        stroke="currentColor"
        stroke-width="0"
        version="1.1"
        viewBox="0 0 16 16"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 3.538c-0.588 0.263-1.222 0.438-1.884 0.516 0.678-0.406 1.197-1.050 1.444-1.816-0.634 0.375-1.338 0.65-2.084 0.797-0.6-0.638-1.453-1.034-2.397-1.034-1.813 0-3.281 1.469-3.281 3.281 0 0.256 0.028 0.506 0.084 0.747-2.728-0.138-5.147-1.444-6.766-3.431-0.281 0.484-0.444 1.050-0.444 1.65 0 1.138 0.578 2.144 1.459 2.731-0.538-0.016-1.044-0.166-1.488-0.409 0 0.013 0 0.028 0 0.041 0 1.591 1.131 2.919 2.634 3.219-0.275 0.075-0.566 0.116-0.866 0.116-0.212 0-0.416-0.022-0.619-0.059 0.419 1.303 1.631 2.253 3.066 2.281-1.125 0.881-2.538 1.406-4.078 1.406-0.266 0-0.525-0.016-0.784-0.047 1.456 0.934 3.181 1.475 5.034 1.475 6.037 0 9.341-5.003 9.341-9.341 0-0.144-0.003-0.284-0.009-0.425 0.641-0.459 1.197-1.038 1.637-1.697z"
        />
      </svg>
    </button>
  </div>
</section>
`;
